code/modules/client/preference_setup/preference_setup.dm

```diff
@@ -117,29 +123,50 @@
 /datum/category_group/player_setup_category/proc/load_character(var/savefile/S)
 	// Load all data, then sanitize it.
 	// Need due to, for example, the 01_basic module relying on species having been loaded to sanitize correctly but that isn't loaded until module 03_body.
+	if (!config.sql_saves || !establish_db_connection(dbcon))
+		for(var/datum/category_item/player_setup_item/PI in items)
+			PI.load_character(S)
+	else
+		// Load every category minus the global
+		handle_sql_loading(SQL_CHARACTER)
+
 	for(var/datum/category_item/player_setup_item/PI in items)
-		PI.load_character(S)
-	for(var/datum/category_item/player_setup_item/PI in items)
-		PI.sanitize_character()
+		PI.sanitize_character(config.sql_saves)
 
 /datum/category_group/player_setup_category/proc/save_character(var/savefile/S)
 	// Sanitize all data, then save it
-	for(var/datum/category_item/player_setup_item/PI in items)
+	for (var/datum/category_item/player_setup_item/PI in items)
 		PI.sanitize_character()
-	for(var/datum/category_item/player_setup_item/PI in items)
-		PI.save_character(S)
+
+	if (!config.sql_saves || !establish_db_connection(dbcon))
+		for (var/datum/category_item/player_setup_item/PI in items)
+			PI.save_character(S)
+	else
+		if (modified && sql_role == SQL_CHARACTER)
+			handle_sql_saving(SQL_CHARACTER)
+			modified = 0
 
 /datum/category_group/player_setup_category/proc/load_preferences(var/savefile/S)
-	for(var/datum/category_item/player_setup_item/PI in items)
-		PI.load_preferences(S)
-	for(var/datum/category_item/player_setup_item/PI in items)
-		PI.sanitize_preferences()
+	if (!config.sql_saves || !establish_db_connection(dbcon))
+		for (var/datum/category_item/player_setup_item/PI in items)
+			PI.load_preferences(S)
+	else
+		handle_sql_loading(SQL_PREFERENCES)
+
+	for (var/datum/category_item/player_setup_item/PI in items)
+		PI.sanitize_preferences(config.sql_saves)
 
 /datum/category_group/player_setup_category/proc/save_preferences(var/savefile/S)
-	for(var/datum/category_item/player_setup_item/PI in items)
+	for (var/datum/category_item/player_setup_item/PI in items)
 		PI.sanitize_preferences()
-	for(var/datum/category_item/player_setup_item/PI in items)
-		PI.save_preferences(S)
+
+	if (!config.sql_saves || !establish_db_connection(dbcon))
+		for (var/datum/category_item/player_setup_item/PI in items)
+			PI.save_preferences(S)
+	else
+		if (modified && sql_role == SQL_PREFERENCES)
```

Comment:
This is unnecessary, save preferences needs to run every single time. Always.
Save character, not so much.